AdobeMediaGallery/Test/Unit/Model/Keyword/Command/GetAssetKeywordsTest.php

```diff
@@ -0,0 +1,140 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\AdobeMediaGallery\Test\Unit\Model\Keyword\Command;
+
+use Magento\AdobeMediaGallery\Model\Keyword\Command\GetAssetKeywords;
+use Magento\AdobeMediaGalleryApi\Api\Data\KeywordInterface;
+use Magento\AdobeMediaGalleryApi\Api\Data\KeywordInterfaceFactory;
+use Magento\Framework\App\ResourceConnection;
+use Magento\Framework\DB\Adapter\AdapterInterface;
+use Magento\Framework\DB\Select;
+use Magento\Framework\Exception\NotFoundException;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+
+class GetAssetKeywordsTest extends TestCase
+{
+    /**
+     * @var GetAssetKeywords
+     */
+    private $sut;
+
+    /**
+     * @var ResourceConnection | MockObject
+     */
+    private $resourceConnectionStub;
+
+    /**
+     * @var KeywordInterfaceFactory | MockObject
+     */
+    private $assetKeywordFactoryStub;
+
+    protected function setUp()
+    {
+        $this->resourceConnectionStub = $this->createMock(ResourceConnection::class);
+        $this->assetKeywordFactoryStub = $this->createMock(KeywordInterfaceFactory::class);
+
+        $this->sut = new GetAssetKeywords(
+            $this->resourceConnectionStub,
+            $this->assetKeywordFactoryStub
+        );
+    }
+
+    public function testCanBeCreated()
+    {
+        $this->assertInstanceOf( GetAssetKeywords::class, $this->sut);
+    }
+
+    public function testFindOne()
+    {
+        $assetIdWithOneKeyword = 1;
+        $oneKeywordQueryResult = ['keywordRawData'];
+        $this->configureResourceConnectionStub($oneKeywordQueryResult);
+        $this->configureAssetKeywordFactoryStub();
+
+        $keywords = $this->sut->execute($assetIdWithOneKeyword);
+
+        $this->assertCount(1, $keywords);
```

Comment:
I think it's better to perform more strict assertion here:
```suggestion
        $this->asserEquals($oneKeywordQueryResult, $keywords);
```

Reply:
I thought about this, but I have finally decided to test only the responsibility of the command. In this case, check the content of the array would be test something that is moked `\Magento\AdobeMediaGallery\Test\Unit\Model\Keyword\Command\GetAssetKeywordsTest::configureAssetKeywordFactoryStub` and this makes the test fragile.